src/react-components/src/helpers/hoc/WithOneChild.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+
+/**
+ * Wrapper component to verify that a component only has one child.
+ * @param Component Component to test, either a React element
+ *   or an object element.
+ */
+const withOneChild = <P extends object>(
+  Component: React.ComponentType<P> | any
+): React.ComponentType<P> => (
+  ({ ...initProps }) => {
+    // It's more likely that the component passed will be a
+    // React element so set these as default.
+    let componentChildren = initProps.children;
+    let componentToRender = Component;
+    let name = Component.name;
+    let props = initProps;
+
+    // In case the component is an object!
+    // E.g, a component passed as a prop.
+    if (Component && typeof Component === "object") {
+      componentChildren = Component.props.children;
+      name = Component.type;
+      props = Component.props;
+      componentToRender = Component.type;
+    }
+
+    // If the passed component is a string, wrap it in a fragment.
+    if (componentChildren && typeof componentChildren === "string") {
+      componentChildren = <>{componentChildren}</>;
```

Comment:
Under what circumstances would the component be a string?

Reply:
If you have a link wrapper component, for example, that can take in `<b>Click me</b>` or just `Click me`. So it would cover the cases where the child is a string or a component.